lib/linguist/heuristics.rb

```diff
@@ -75,6 +75,11 @@ def call(data)
     end
 
     # Common heuristics
+    CRegex = Regexp.union(
+        /\b(char|double|short|int|long|float|unsigned|void)\s+\w+\s*\(\s*void\s*\)/,
+        /\([\s\w\*]+restrict\s+/,
+        /\b(char|short|int|long|float|unsigned)\s+\w+\s*\[\]\s*;/
```

Comment:
What makes you think this last heuristic rule is only valid in C?

Reply:
@pchaigno Flexible struct parameters are not valid C++, see https://stackoverflow.com/a/4413035/5129504

Reply:
`.h` is also [an Objective-C extension](https://github.com/github/linguist/blob/5363e045bb2728eec05ff2225a87e742e0bff586/lib/linguist/languages.yml#L3067).

Reply:
… and precisely why the tests are currently failing, suggesting further refinement is needed.